homeassistant/core.py

```diff
@@ -1457,7 +1509,8 @@ def _async_fire(
                 except Exception:  # pylint: disable=broad-except
                     _LOGGER.exception("Error running job: %s", job)
             else:
-                self._hass.async_add_hass_job(job, event)
+                # pylint: disable-next=protected-access
+                self._hass._async_add_hass_job(job, event)
```

Comment:
We only have 3 places left where this is called. We should try to get rid of them so we can deprecate this as well

```python
homeassistant/components/matrix/__init__.py:            EVENT_HOMEASSISTANT_START, handle_startup, run_immediately=False
homeassistant/components/rfxtrx/__init__.py:            dr.EVENT_DEVICE_REGISTRY_UPDATED, _updated_device, run_immediately=False
homeassistant/helpers/event.py:        run_immediately=False,
```

Reply:
event can be switched after <https://github.com/home-assistant/core/pull/115160>

If we can fix the other two we can deprecate `run_immediately=False` and get rid of this completely in a future PR